diff --git a/mac-release.bash b/mac-release.bash
index 1111111..2222222 100644
--- a/mac-release.bash
+++ b/mac-release.bash
@@ -8,6 +8,9 @@
 
 THIS_DIR=$(pwd)
 
+# Added for Node.js addon (dlopen) compatibility
+LIBBUN_PIC_FLAG="-fPIC"
+LIBBUN_TLS_FLAG="-ftls-model=global-dynamic"
+
 # Set default values for environment variables that are not set.
 CMAKE_C_COMPILER=${CMAKE_C_COMPILER:-clang-19}
 CMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER:-clang++}
@@ -40,8 +43,8 @@ cmake \
     -DPORT="JSCOnly" \
     -DENABLE_STATIC_JSC=ON \
     -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
     -DENABLE_BUN_SKIP_FAILING_ASSERTIONS=ON \
     -DUSE_THIN_ARCHIVES=OFF \
     -DENABLE_FTL_JIT=ON \
     -DCMAKE_C_COMPILER="$CMAKE_C_COMPILER" \
     -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
     -DCMAKE_CXX_COMPILER="$CMAKE_CXX_COMPILER" \
-    -DCMAKE_C_FLAGS="-g $CMAKE_C_FLAGS -ffile-prefix-map=$THIS_DIR/Source=vendor/WebKit/Source" \
-    -DCMAKE_CXX_FLAGS="-g $CMAKE_CXX_FLAGS -ffile-prefix-map=$THIS_DIR/Source=vendor/WebKit/Source" \
+    -DCMAKE_C_FLAGS="-g $CMAKE_C_FLAGS $LIBBUN_PIC_FLAG $LIBBUN_TLS_FLAG -ffile-prefix-map=$THIS_DIR/Source=vendor/WebKit/Source" \
+    -DCMAKE_CXX_FLAGS="-g $CMAKE_CXX_FLAGS $LIBBUN_PIC_FLAG $LIBBUN_TLS_FLAG -ffile-prefix-map=$THIS_DIR/Source=vendor/WebKit/Source" \
     -DENABLE_MALLOC_HEAP_BREAKDOWN=$ENABLE_MALLOC_HEAP_BREAKDOWN \
     -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \

