diff --git a/windows-release.ps1 b/windows-release.ps1
index 1111111..2222222 100644
--- a/windows-release.ps1
+++ b/windows-release.ps1
@@ -168,6 +168,9 @@ Write-Host ":: Configuring WebKit"
 
 $env:PATH = $PathWithPerl
 
+# Added for Node.js addon (dlopen) compatibility
+$LibbunTlsFlag = "-Xclang -ftls-model=global-dynamic"
+
 $env:CFLAGS = "/Zi"
 $env:CXXFLAGS = "/Zi"
 
@@ -201,10 +204,10 @@ cmake -S . -B $WebKitBuild `
     "-DICU_LIBRARY=${ICU_STATIC_LIBRARY}" `
     "-DICU_INCLUDE_DIR=${ICU_STATIC_INCLUDE_DIR}" `
     "-DCMAKE_C_COMPILER=clang-cl" `
     "-DCMAKE_CXX_COMPILER=clang-cl" `
-    "-DCMAKE_C_FLAGS_RELEASE=/Zi /O2 /Ob2 /DNDEBUG  " `
-    "-DCMAKE_CXX_FLAGS_RELEASE=/Zi /O2 /Ob2 /DNDEBUG  -Xclang -fno-c++-static-destructors " `
-    "-DCMAKE_C_FLAGS_DEBUG=/Zi /FS /O0 /Ob0 " `
-    "-DCMAKE_CXX_FLAGS_DEBUG=/Zi /FS /O0 /Ob0 -Xclang -fno-c++-static-destructors " `
+    "-DCMAKE_C_FLAGS_RELEASE=/Zi /O2 /Ob2 /DNDEBUG $LibbunTlsFlag " `
+    "-DCMAKE_CXX_FLAGS_RELEASE=/Zi /O2 /Ob2 /DNDEBUG $LibbunTlsFlag -Xclang -fno-c++-static-destructors " `
+    "-DCMAKE_C_FLAGS_DEBUG=/Zi /FS /O0 /Ob0 $LibbunTlsFlag " `
+    "-DCMAKE_CXX_FLAGS_DEBUG=/Zi /FS /O0 /Ob0 $LibbunTlsFlag -Xclang -fno-c++-static-destructors " `
     -DENABLE_REMOTE_INSPECTOR=ON `
     "-DCMAKE_MSVC_RUNTIME_LIBRARY=${CmakeMsvcRuntimeLibrary}" `

